### DISCUSSION TITLE
Video format 16 bit

### DISCUSSION THREAD

**Discussion by david3a** [vsf-tv/gccg-api#32] [General]
We discussed using planar 16 bit video, microsoft specify a FOURCC code P216 but this uses two planes, a Y plane then a UV plane, details here

[https://learn.microsoft.com/en-us/windows/win32/medfound/10-bit-and-16-bit-yuv-video-formats#p216-and-p210](url)

The EBU DMF document [https://tech.ebu.ch/files/live/sites/tech/files/shared/white_papers/EBU_White_Paper_The_Dynamic%20Media_Facility_Reference_Architecture.pdf](url)

defines 16 video as planar but as strides of Y then U (Cb) then V (Cr)

My preference would be Y plane, followed by U plane followed by V plane

**Comment by david3a** [comment]
Should have also said we should support YUVA as well where A would be a full resolution Alpa channel, so for a simple example with a resolution of 16 x 8 (its an example!)

YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
YYYYYYYYYYYYYYYY
UUUUUUUU
UUUUUUUU
UUUUUUUU
UUUUUUUU
UUUUUUUU
UUUUUUUU
UUUUUUUU
UUUUUUUU
VVVVVVVV
VVVVVVVV
VVVVVVVV
VVVVVVVV
VVVVVVVV
VVVVVVVV
VVVVVVVV
VVVVVVVV

with Alpha if provided

AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAA

EBU DMF would propose

YYYYYYYYYYYYYYYY
UUUUUUUU
VVVVVVVV
YYYYYYYYYYYYYYYY
UUUUUUUU
VVVVVVVV

etc

**Comment by gkelly1** [comment]
We should consider YUV422p10le Format as it is fairly well supported today. It uses the lower bits + padding in 16b format. 


![YUV422p10le](https://github.com/user-attachments/assets/7b5b87ba-96f1-4ab1-bb25-cf32c9a2b1ac)

I believe the P210/P216 is similar to Y210/Y216 just Planar version. Y210/Y216 use padding in the lower bits + pixel data. This is typically used in accelerators.


![y210](https://github.com/user-attachments/assets/8148a351-fdf3-47ab-be47-1e0634ff3b9b)

We should also consider RGBA/ARGB type format for Graphics
